### ISSUE TITLE
AntiVPN (Proxy) Bot via Query

### ISSUE THREAD

**Issue by SkyCuteFX** [xDefcon/sinusbot-scripts#31]
would it be possible that you write a query because then TS3Hook can still scan. best php?

**Comment by SkyCuteFX** [comment]
edit: because the query can read the IP despite TS3Hook.

**Comment by xDefcon** [comment]
Yes, it would be possible, you can achieve that with any kind of language that can connect to TS3 via ServerQuery, Java and PHP would be the most suitable.

Here's the algorithm to detect if someone is using TS3Hook to hide his IP address (very simple):
When a client connects, check for his IP address, if not shown then repeat this check at random intervals, after X fails, then you know that the client is hiding his IP address.

With AntiProxy and SinusBot you should be 100% able to detect who hides his IP address, aren't you?

**Comment by SkyCuteFX** [comment]
I would recommend you to write a Querybot because through the Querybot everything can be read even with TS3Hook thus one would then have to ban / 100 percentiege schongs VPN and / or proxies Because the TS3Hook bypass bugt mostly. and so people can use the TS3Hook continue to use it but the IP is read by query anyway. 🍡

**Comment by JustAB0x** [comment]
There is a repo i found the other day .Code seems clean. Didn't test it tho.
@xDefcon Hope its okay if i leave the link here
https://github.com/dying/Bot-Query-TeamSpeak-PHP-AntiVPN-AntiProxy

**Comment by xDefcon** [comment]
ServerQuery support will be probably released in the next Script version

**Comment by SkyCuteFX** [comment]
This is cool, I'm excited!

Short question where do you get your database for the IP addresses (VPN/Bad IP and Internet Provider)

**Comment by xDefcon** [comment]
> This is cool, I'm excited!
> 
> Short question where do you get your database for the IP addresses (VPN/Bad IP and Internet Provider)

Here I link you some reputable ones where you can get all the DB you need (IP to ISP, Country, City etc.): 
 
- https://www.maxmind.com/en/home
- https://www.ip2location.com/
- https://db-ip.com/ (they provide both API and DB)
